adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralInterstitialRenderer.m

```diff
@@ -0,0 +1,121 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralInterstitialRenderer.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+#include <stdatomic.h>
+#import <MTGSDKNewInterstitial/MTGSDKNewInterstitial.h>
+#import <MTGSDKNewInterstitial/MTGNewInterstitialBidAdManager.h>
+
+@interface GADMAdapterMintegralInterstitialRenderer ()<GADMediationInterstitialAd,MTGNewInterstitialBidAdDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralInterstitialRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationInterstitialLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// Ad configuration for the ad to be loaded.
+    GADMediationInterstitialAdConfiguration *_adConfiguration;
+    
+    /// The Mintegral interstitial ad.
+    MTGNewInterstitialBidAdManager *_interstitialAd;
+    
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationInterstitialAdEventDelegate> _adEventDelegate;
+}
+
+- (void)renderInterstitialForAdConfiguration:(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)
+                                                 completionHandler{
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationInterstitialLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationInterstitialAdEventDelegate>(
+                                                                        _Nullable id<GADMediationInterstitialAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    _interstitialAd = [[MTGNewInterstitialBidAdManager alloc]initWithPlacementId:placementId unitId:adUnitId delegate:self];
+    [_interstitialAd loadAdWithBidToken:adConfiguration.bidResponse];
```

Comment:
Can you remind me again if Mintegral SDK verify bid token internally?

Reply:
Mintegral Ad SDK validates bid tokens internally.